core/src/sigverify.rs

```diff
@@ -342,6 +387,60 @@ mod tests {
         assert_eq!(sig_len, 1);
     }
 
+    #[test]
+    fn test_untrustworthy_sigs() {
+        let required_num_sigs = 14 as u32;
+        let actual_num_sigs = 5 as u32;
+
+        let message = Message {
+            header: MessageHeader {
+                num_required_signatures: required_num_sigs as u8,
+                num_credit_only_signed_accounts: 12,
+                num_credit_only_unsigned_accounts: 11,
+            },
+            account_keys: vec![],
+            recent_blockhash: Hash::default(),
+            instructions: vec![],
+        };
+        let mut tx = Transaction::new_unsigned(message);
+        tx.signatures = vec![Signature::default(); actual_num_sigs as usize];
+        let _tx_bytes = serialize(&tx).unwrap();
+        let packet = sigverify::make_packet_from_transaction(tx.clone());
+
+        let (trustworthy, sig_len, _sig_start, _msg_start_offset, _pubkey_offset) =
+            sigverify::do_get_packet_offsets(&packet, 0);
+
+        assert_eq!(trustworthy, false);
+        assert_eq!(sig_len, actual_num_sigs);
+    }
+
+    #[test]
+    fn test_large_sigs() {
+        // use any large number to be misinterpreted as 2 bytes when decoded as short_vec
+        let required_num_sigs = 214 as u32;
+        let actual_num_sigs = 5 as u32;
+
+        let message = Message {
+            header: MessageHeader {
+                num_required_signatures: required_num_sigs as u8,
+                num_credit_only_signed_accounts: 12,
+                num_credit_only_unsigned_accounts: 11,
+            },
+            account_keys: vec![],
+            recent_blockhash: Hash::default(),
+            instructions: vec![],
+        };
+        let mut tx = Transaction::new_unsigned(message);
+        tx.signatures = vec![Signature::default(); actual_num_sigs as usize];
+        let _tx_bytes = serialize(&tx).unwrap();
+        let packet = sigverify::make_packet_from_transaction(tx.clone());
+
+        let (_trustworthy, _sig_len, _sig_start, _msg_start_offset, pubkey_offset) =
+            sigverify::do_get_packet_offsets(&packet, 0);
+
+        assert_eq!(pubkey_offset, 325);
```

Comment:
Can you show where this magic number comes from? Maybe something like this:
https://github.com/solana-labs/solana/blob/322fcea6e51a600d8cfcad3f68c4866c094f553c/sdk/src/transaction.rs#L422

Reply:
You're right to blame my laziness... :p 003521a74e9b4b52aadadb9633d42b9cadbd36af